src/components/pages/HeadsUpDisplay/CurrentActivity.js

```diff
@@ -1,117 +1,101 @@
-import React, { useState, useEffect } from 'react';
-
-export const CountDownTimer = () => {
-  // This function will calculate difference between current and target times
-  const calculateTimeRemaining = () => {
-    let currTime = new Date();
-    // let currTime = new Date("2021-08-08T23:59:59");
-    // let currTime = new Date("2021-08-09T00:00:00");
-    // console.log("currTime", currTime);
-
-    // If the day of week is Saturday, countdown will be 48 hours, otherwise, 24 hours
-    let endTime = new Date().setHours(24, 0, 0, 0);
-    // console.log("endTime", convertTimeFormat(endTime));
-    // currTime.getDay() === 6
-    //   ? new Date().setHours(48, 0, 0, 0)
-    //   : new Date().setHours(24, 0, 0, 0);
-
-    // let endTime = new Date("2021-08-09T00:00:00");
-
-    let timeDiff = endTime - currTime;
-
-    // console.log("timeDiff", convertTimeFormat(timeDiff));
-
-    return timeDiff;
-  };
-
-  // This function will convert the format for our time difference to HH:MM:SS
-  const convertTimeFormat = time => {
-    let timeVar = parseInt(time, 10);
-
-    let hours = Math.floor(timeVar / (1000 * 60 * 60));
-    let minutes = Math.floor((timeVar / (1000 * 60)) % 60);
-    let seconds = Math.floor(timeVar / 1000) % 60;
-
-    return [hours, minutes, seconds]
-      .map(v => (v < 10 ? '0' + v : v))
-      .filter((v, i) => v !== '00' || i > 0)
-      .join(':');
+import React, { useState, useEffect } from "react";
+
+export const CurrentActivity = () => {
+
+  const [day, setDay] = useState(1);
+  const [activity, setActivity] = useState("");
+
+  // This function will change current day (0-13) based on previous day
+  // Will come back and refactor this to make it dry later
+  const currDay = (day) => {
+    let today = new Date().getDay();
+    setDay(today);
+    if (today === 0) {
+      if (day === 13) {
+        setDay(0);
+        return;
+      } else if (day === 6) {
+        setDay(7);
+        return;
+      }
+    } else if (today === 1) {
+      if (day === 0) {
+        setDay(1);
+        return;
+      } else if (day === 7) {
+        setDay(8);
+        return;
+      }
+    } else if (today === 2) {
+      if (day === 1) {
+        setDay(2);
+        return;
+      } else if (day === 8) {
+        setDay(9);
+        return;
+      }
+    } else if (today === 3) {
+      if (day === 2) {
+        setDay(3);
+        return;
+      } else if (day === 9) {
+        setDay(10);
+        return;
+      }
+    } else if (today === 4) {
+      if (day === 3) {
+        setDay(4);
+        return;
+      } else if (day === 10) {
+        setDay(11);
+        return;
+      }
+    } else if (today === 5) {
+      if (day === 4) {
+        setDay(5);
+        return;
+      } else if (day === 11) {
+        setDay(12);
+        return;
+      }
+    } else if (today === 6) {
+      if (day === 5) {
+        setDay(6);
+        return;
+      } else if (day === 12) {
+        setDay(13);
+        return;
+      }
+    }
```

Comment:
Love the execution here. There maybe a way to make it more readable, can you think of anyways to extract some of the logic into a function?

Reply:
I have an idea how I can make this more readable, my plan was to get the feature in and then go back and reconfigure.